driver-core/src/main/com/mongodb/operation/AsyncChangeStreamBatchCursor.java

```diff
@@ -101,7 +120,11 @@ public int getBatchSize() {
 
     @Override
     public boolean isClosed() {
-        return wrapped.isClosed();
+        boolean closed = false;
+        synchronized (this) {
```

Comment:
You can simplify this, as in AsyncQueryBatchCursor:

        synchronized (this) {
            return isClosed;
        }

Reply:
Done